python/ray/_private/utils.py

```diff
@@ -1286,8 +1286,8 @@ def get_directory_size_bytes(path: Union[str, Path] = ".") -> int:
     for dirpath, dirnames, filenames in os.walk(path):
         for f in filenames:
             fp = os.path.join(dirpath, f)
-            # skip if it is a symbolic link
-            if not os.path.islink(fp):
+            # skip if it is a symbolic link or a .pyc file
+            if not os.path.islink(fp) and not f.endswith(".pyc"):
```

Comment:
I think this is also a walk around. @architkulkarni Maybe you can enhance this in future PR? One alternative is that record the size when download.

Reply:
Makes sense, I created issue #23186 to track the improvement.  @rkooo567 we used to record the size when downloading in an early iteration of the URI cache PR, but I think we got rid of that logic to make the code simpler.  I think it's reasonable to add it back, what do you think?  I don't think it increases the complexity too much.